### ISSUE TITLE
Query the domain name under test for CNAME

### ISSUE THREAD

**Issue by mattias-p** [zonemaster/zonemaster#797]
At the start of a test Zonemaster tries to find out the NS for a given domain name from its parent zone. If the domain name is an alias there are no NS and the test is aborted.

Today when this happens the user is left to his or her own devices to figure out that the provided domain name is an alias.

We should extend BASIC02 to also make a CNAME query for the given domain name and report it as an error if such a record is found. The error should include a parameter for the canonical name in the CNAME record.

This way the user would get immediate information about the domain name being an alias as opposed to a zone. Additionally Zonemaster would be able to detect situations where a delegation from the parent is combined with an alias.

This came up in zonemaster/zonemaster-engine#539.

**Comment by matsduf** [comment]
No, I do not think that we should clutter [Basic02] with such a test, but I think it could make sense to extend [Basic03] which is only run if Basic02 fails.

[Basic02]:https://github.com/zonemaster/zonemaster/blob/master/docs/specifications/tests/Basic-TP/basic02.md
[Basic03]:https://github.com/zonemaster/zonemaster/blob/master/docs/specifications/tests/Basic-TP/basic03.md

**Comment by mattias-p** [comment]
If we add the CNAME check to Basic03 instead of Basic02 Zonemaster won't be able to detect the error when both NS and CNAME is present in the parent.

**Comment by matsduf** [comment]
I think that is a separate test (to discover CNAME and NS at parent). That is a candidate for testing in another test case. The use case that you refer to ("the domain name is an alias there are no NS") is better to be handled in Basic03 which is meant for cases like that.

**Comment by matsduf** [comment]
There is no reason to explicitly query for CNAME in expected delegation node since CNAME will be returned when querying for other record types. NS and CNAME cannot be combined in the same node.

The proposed update of Basic01 (#1082) checks for CNAME in the reply and will in the combination with the proposed update of Basic02 (#1085) cover the need from this issue.
